### PR TITLE
easyloggingpp: Add package

### PR DESCRIPTION
**Pull request by VolkerChristian**
Maintainer: me@vchrist.at
Compile tested: OpenWRT-23.05-rc2, arm_cortex-a7_neon-vfpv4, mips_24kc
Run tested: arm_cortex-a7_neon-vfpv4 (Linksys MR8300), mips_24kc (tplink_archer-a7), OpenWrt 23.05-rc2)

Description:
Easylogging++ is single header efficient logging library for C++ applications. It is extremely powerful, highly extendable and configurable to user's requirements.

Signed-off-by: Volker Christian <me@vchrist.at>

### CODE REVIEW

File: libs/easyloggingpp/Makefile [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+PKG_VERSION:=9.97.0
+PKG_RELEASE:=1
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE_URL:=https://github.com/abumq/easyloggingpp.git
+PKG_SOURCE_VERSION:=v9.97.0
+PKG_SOURCE_DATE:=2020-12-25
+
+PKG_MIRROR_HASH:=9acdb77ca418b07feef4ffc4ae1858c0a11f9e44c44a0eff21d3e278844e4dc1
```
Comment (1715173329) [inline libs/easyloggingpp/Makefile:13]: use codeload instead, for example:
https://github.com/openwrt/packages/blob/1cb2590c1743eeb4c357b1f0d7e3fb47b3640ae6/utils/yq/Makefile#L7-L9

File: libs/easyloggingpp/Makefile [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+PKG_MAINTAINER:=Volker Christian <me@vchrist.at>
+
+PKG_LICENSE:=MIT
+PKG_LICENSE_FILES:=LICENSE
```
Comment (1715173329) [inline libs/easyloggingpp/Makefile:18]:
```suggestion
PKG_MAINTAINER:=Volker Christian <me@vchrist.at>
PKG_LICENSE:=MIT
PKG_LICENSE_FILES:=LICENSE
```

File: libs/easyloggingpp/Makefile [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+PKG_BUILD_DEPENDS:=toolchain
```
Comment (1715173329) [inline libs/easyloggingpp/Makefile:20]:
```suggestion
CMAKE_INSTALL:=1
```

- toolchain is always built, no need to specify
- use CMAKE_INSTALL to simplify InstallDev

File: libs/easyloggingpp/Makefile [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+endef
+
+define Build/InstallDev
+	$(INSTALL_DIR) $(1)/usr/include
+	$(INSTALL_DATA) $(PKG_INSTALL_DIR)/usr/include/easylogging++.h $(1)/usr/include
+	$(INSTALL_DATA) $(PKG_INSTALL_DIR)/usr/include/easylogging++.cc $(1)/usr/include
+	$(INSTALL_DIR) $(1)/usr/lib/pkgconfig
+	$(INSTALL_DATA) $(PKG_INSTALL_DIR)/usr/share/pkgconfig/easyloggingpp.pc $(1)/usr/lib/pkgconfig
+endef
+
```
Comment (1715173329) [inline libs/easyloggingpp/Makefile:46]:
```suggestion
```

CMAKE_INSTALL does it already

**Review by 1715173329 — CHANGES_REQUESTED** [review]
And I noticed the latest release was updated about 3 years ago, is it really active?

**Comment by VolkerChristian** [comment]
Thank you for the hint to use codeload and CMAKE_INSTALL! I have updated the pull request accordingly.

As for the activity of the project: Upstream has at least made some commits two month ago. But you are right - there is no new release since two and a half year ago.
Nevertheless, the software I want to bring to OpenWRT ([SNode.C](https://github.com/VolkerChristian/snode.c) and [MQTTBroker](https://github.com/VolkerChristian/mqttbroker)) relies heavily on easyloggingpp.

File: libs/easyloggingpp/Makefile [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+PKG_NAME:=easyloggingpp
+PKG_VERSION:=9.97.0
+PKG_RELEASE:=1
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://codeload.github.com/abumq/easyloggingpp/tar.gz/v$(PKG_VERSION)?
+PKG_HASH:=9110638e21ef02428254af8688bf9e766483db8cc2624144aa3c59006907ce22
+PKG_SOURCE_VERSION:=v9.97.0
+PKG_SOURCE_DATE:=2020-12-25
```
Comment (1715173329) [inline libs/easyloggingpp/Makefile:11]:
```suggestion
```
unused variables

**Review by 1715173329 — APPROVED** [review]
otherwise LGTM - Thanks!

**Pull request merged**